data/src/main/java/com/linkedin/data/schema/validator/DataSchemaAnnotationValidator.java

```diff
@@ -424,6 +424,7 @@ public void callback(List<String> path, DataSchema schema)
         {
           Constructor<? extends Validator> ctor = clazz.getConstructor(DataMap.class);
           DataMap configDataMap = (DataMap) config;
+          configDataMap.makeReadOnly();
```

Comment:
Since this is the single line change here, I am out of context, do you mind say why we need this readOnly?

Reply:
If a DataMap is marked readonly we don't even add change listeners when it is wrapped by record template.
readOnly => will not change => no need for change listener.

So making it readonly here ensures that validator instances do not create a change listener for each field being validated. While the fix in CheckedMap ensures we GC change listeners correctly, this change avoids creating a large list of listeners in the first place.

Reply:
Added comments